sdk/core/core-paging/src/models.ts

```diff
@@ -31,3 +31,22 @@ export interface PagedAsyncIterableIterator<T, PageT = T[], PageSettingsT = Page
    */
   byPage: (settings?: PageSettingsT) => AsyncIterableIterator<PageT>;
 }
+
+/**
+ * An interface that describes how to communicate with the service.
+ */
+export interface PagedResult<TPage, TPageSettings = PageSettings> {
+  /**
+   * Link to the first page of results.
+   */
+  link: string;
+  /**
+   * A method that returns a page of results.
+   */
+  getPage: (link: string, maxPageSize?: number) => Promise<{ page: TPage; nextLink?: string }>;
```

Comment:
same naming thoughts here
```suggestion
  getPage: (pageUrl: string, maxPageSize?: number) => Promise<{ page: TPage; nextLinkUrl?: string }>;
```